### DISCUSSION TITLE
Cog error

### DISCUSSION THREAD

**Discussion by diavolicchio** [Rapptz/discord.py#7097] [Q&A]
Hello! I'm trying to make a welcome cog in discord.py but I keep getting this error:
module 'discord.ext.commands.cog' has no attribute 'listener'
I'm using discord.py rewrite and everything is up-to-date, any idea on how to fix this?

**Comment by imvaskel** [comment]
can you provide the code you are using?

**Reply by diavolicchio** [reply]
Alr a sec

**Reply by diavolicchio** [reply]
class Welcome(commands.Cog, name='Welcomer') :

    def __init__(self,bot):
        self.bot = bot

    @commands.Cog.listener()
    async def on_member_join(self,member):
        channel = self.bot.get_channel(844100431155822602)
        embed=discord.Embed(description=f'Un invasore,{member.mention},si è infiltrato nel server!')
        embed.set_image(url="https://cdn.discordapp.com/attachments/843969304164827199/856948707852812298/Fermo-1.png")
    
    @commands.cog.listener()
    async def on_member_remove(self,member):
        channel = self.bot.get_channel(844100573854564383)
        embed = discord.Embed(description=f'{member.name} è appena uscito dal server! Che traditore!')

def setup(bot):
    bot.add_cog(Welcome(bot))

**Reply by esmaycat** [reply]
You are doing `@commands.cog.listener()`, `commands.cog` is the **module**. It should be `@commands.Cog.listener()`, here `commands.Cog` is the **class**.

**Reply by diavolicchio** [reply]
Oh my god I didn't notice it, thanks!
